### PR TITLE
Icon component for customizable SVG icons

### PR DESCRIPTION
**Pull request by strouhaljan**
JIRA: BB-2840

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2616/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4250/ : SUCCESS in 15m 23s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2779/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4411/ : SUCCESS in 12m 28s

**Comment by strouhaljan** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/728/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2784/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/728/ : UNSTABLE in 21m 28s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2790/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4421/ : SUCCESS in 13m 51s

**Comment by strouhaljan** [comment]
extended test

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2798/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4426/ : FAILURE in 6m 36s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2810/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4436/ : FAILURE in 3m 01s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2812/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4438/ : FAILURE in 6m 20s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2815/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4441/ : SUCCESS in 13m 29s

**Comment by strouhaljan** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/732/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2817/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/732/ : UNSTABLE in 21m 45s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2818/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4443/ : SUCCESS in 13m 42s

**Comment by strouhaljan** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/733/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2819/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/733/ : SUCCESS in 22m 03s

File: libs/sdk-ui-kit/src/Icon/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        case "Refresh":
+        case "DrillDown":
+        case "DrillToDashboard":
+        case "DrillToInsight": {
+            const IconComponent = Icons[name];
+            return <IconComponent name={name} color={color} colorPalette={colorPalette} />;
+        }
+    }
+
+    iconNotFoundCheck(name);
```
Comment (kandl) [inline libs/sdk-ui-kit/src/Icon/Icon.tsx:27]: I would move it to default switch statement (as it is likely more proper place to handle exhaustive check).

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by strouhaljan** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2830/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/734/ : FAILURE in 3m 12s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2827/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4452/ : FAILURE in 2m 53s

**Comment by strouhaljan** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2831/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/735/ : FAILURE in 3m 46s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2833/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4459/ : SUCCESS in 12m 21s

**Comment by strouhaljan** [comment]
extended test

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/736/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2834/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/736/ : SUCCESS in 16m 46s

File: libs/sdk-ui-tests/stories/visual-regression/kit/Icon/Icon.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const IconTest: React.FC = () => {
+    return (
+        <div className="library-component screenshot-target">
+            <IconWrapper name="Refresh" color="#f00" />
+            <IconWrapper name="DrillDown" color="#0f0" />
+            <IconWrapper name="DrillToDashboard" color="#00f" />
+            <IconWrapper name="DrillToInsight" color="#f00" />
+        </div>
+    );
+};
```
Comment (scavnickyj) [inline libs/sdk-ui-tests/stories/visual-regression/kit/Icon/Icon.tsx:27]: Some icons are missing.

File: libs/sdk-ui-kit/src/Icon/icons/DrillToDashboard.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * @internal
+ */
+export const DrillToDashboard: React.FC<IIconProps> = ({ color }) => {
+    return (
+        <svg width="16px" height="16px" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg">
+            <g id="Artboard" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
+                <polygon
+                    id=""
```
Comment (scavnickyj) [inline libs/sdk-ui-kit/src/Icon/icons/DrillToDashboard.tsx:14]: What is this character? Do we even need those externally generated attributes in icons?

File: libs/sdk-ui-tests/stories/visual-regression/kit/Icon/Icon.tsx (hunk 2 above)
Comment (strouhaljan) [reply libs/sdk-ui-tests/stories/visual-regression/kit/Icon/Icon.tsx:27]: Adding those just now :)

File: libs/sdk-ui-kit/src/Icon/icons/DrillToDashboard.tsx (hunk 3 above)
Comment (strouhaljan) [reply libs/sdk-ui-kit/src/Icon/icons/DrillToDashboard.tsx:14]: I guess we don't. I already made a bit of cleanup within SVGs and was worried to not mess with its look. To answer your question, this character is just a copy paste from the original.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2836/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4461/ : SUCCESS in 20m 26s

**Comment by strouhaljan** [comment]
extended test

File: libs/sdk-ui-tests/stories/visual-regression/kit/Icon/Icon.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import React from "react";
+import { storiesOf } from "@storybook/react";
+import { UiKit } from "../../../_infra/storyGroups";
+import { withScreenshot } from "../../../_infra/backstopWrapper";
+
+import "@gooddata/sdk-ui-kit/styles/css/main.css";
+import "./styles.scss";
+
+const IconWrapper: React.FC<IIconProps> = ({ name, color }) => (
+    <div className={"icon-wrapper"}>
```
Comment (scavnickyj) [inline libs/sdk-ui-tests/stories/visual-regression/kit/Icon/Icon.tsx:12]: No need for `{}` brackets here.

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/737/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2838/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/737/ : UNSTABLE in 20m 25s

**Review by scavnickyj — APPROVED** [review]

**Comment by scavnickyj** [comment]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1080

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/835/ : SUCCESS in 24s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1099/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4467/ : SUCCESS in 12m 28s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/207/ : SUCCESS in 8m 13s (non-voting)

**Pull request merged**